TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -303,6 +303,21 @@ public class VehicleBehaviorManager : AbstractCustomManager, IVehicleBehaviorMan
 					//}
 #endif
 
+					// Check if turning in the preferred direction, and if turning while it's red is allowed
+					if (stopCar && sqrVelocity <= TrafficPriorityManager.MAX_SQR_STOP_VELOCITY && JunctionRestrictionsManager.Instance.IsTurnOnRedAllowed(prevPos.m_segment, isTargetStartNode)) {
```

Comment:
Missing the "isRecklessDriver" check.

I was thinking about this before, and I think it'd be a better idea for us to use MAX_SQR_YIELD_VELOCITY for regular drivers and then for reckless drivers to have them not stop at all, as that's more true to real life (who makes a full stop at a red light to turn right anyways, let's be honest here).

Reply:
isRecklessDriver does not need to be checked here: They ignore traffic lights anyway. Changing max speed to yield velocity.